mapie/conformity_scores/residual_conformity_scores.py

```diff
@@ -132,3 +141,275 @@ def get_estimation_distribution(
+
+        calres_indexes : NDArray
+            Indexes used for the training of the estimaot and the calibration.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray]
+            - indexes needed for the calibration.
+            - indexes used for the training of the base estimator.
+        """
+        (X_res_indexes,
+         X_cal_indexes,
+         y_res_indexes,
+         y_cal_indexes) = train_test_split(
+            calres_indexes,
+            calres_indexes,
+            test_size=self.split_size,
+            random_state=self.random_state,
+        )
+
+        residuals = np.abs(np.subtract(
+            y[y_res_indexes],
+            y_pred[y_res_indexes]
+        ))
+        residual_estimator_targets = np.log(np.maximum(
+            residuals,
+            np.full(residuals.shape, self.eps)
+        ))
+
+        residual_estimator_ = residual_estimator_.fit(
+            X[X_res_indexes],
+            residual_estimator_targets
+        )
+
+        cal_index = X_cal_indexes
+        train_index = list(set(np.arange(y_pred.shape[0])) - set(cal_index))
+
+        self.residual_estimator_ = residual_estimator_
+
+        return cal_index, np.array(train_index)
+
+    def get_signed_conformity_scores(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        y_pred: ArrayLike
+    ) -> NDArray:
+        """
+        Computes the signed conformity score = (y - y_pred) / r_pred.
+        r_pred being the predicted residual (y - y_pred) of the estimator.
+        It is calculated by a model (``residual_estimator_``) that learns
+        to predict this residual.
+
+        The learning is done with the log of the residual and later we
+        use the exponential of the prediction to avoid negative values.
+        """
+        X, y, y_pred, self.residual_estimator_ = self._check_parameters(
+            X, y, y_pred
+        )
+        calres_indexes = np.argwhere(
+                            np.logical_not(np.isnan(y_pred))
+                        ).reshape((-1, ))
+        if not self.prefit:
+            cal_indexes, train_indexes = self._fit_residual_estimator(
+                clone(self.residual_estimator_), X, y, y_pred, calres_indexes
+            )
+        else:
+            cal_indexes = calres_indexes
+            train_indexes = np.argwhere(np.isnan(y_pred)).reshape((-1,))
+
+        normalizer = np.maximum(
+            np.exp(self.residual_estimator_.predict(X[cal_indexes])),
+            self.eps
+        )
+        signed_conformity_scores = np.divide(
+            np.abs(np.subtract(y[cal_indexes], y_pred[cal_indexes])),
+            normalizer
+        )
+
+        # reconstruct array with nan and conformity scores
+        complete_signed_cs = np.zeros(y_pred.shape)
+        complete_signed_cs[cal_indexes] = signed_conformity_scores
+        complete_signed_cs[train_indexes] = np.full(
+            (train_indexes.shape[0],),
+            np.nan
+        )
```

Comment:
QUESTION: does this work?
```suggestion
        complete_signed_cs[train_indexes] = np.nan
```

Reply:
No it does not work, that's why I did it with np.full

Reply:
What was the error you encountered?

Reply:
`np.nan` is not the same shape of `complete_signed_cs[train_indexes]`

Reply:
I've tried to reproduce this calculation on my machine and I haven't encountered any errors. Can you share me your error?

Reply:
Done